[compile error] no member named 'size' in 'std::shared_ptr<const pcl::PointCloud<pcl::PointXYZ>>'

<!--- WARNING: This is an issue tracker. Before opening a new issue make sure you read https://github.com/PointCloudLibrary/pcl/blob/master/CONTRIBUTING.md#using-the-issue-tracker. -->
**Describe the bug**

```c++
float density = pcl::getMeanPointDensity<pcl::PointXYZ>(cloud, 0.2);
```

```bash
pcl-1.12/pcl/registration/impl/ia_fpcs.hpp:56:31: error: no member named 'size' in 'std::shared_ptr<const pcl::PointCloud<pcl::PointXYZ>>'
  const std::size_t s = cloud.size();
```

**Context**

What are you trying to accomplish? Providing context helps us come up with a solution that is most useful in the real world

**Expected behavior**

A clear and concise description of what you expected to happen.

**Current Behavior**

What happens instead of the expected behavior?

**To Reproduce**

Provide a link to a live example, or an unambiguous set of steps to reproduce this bug. A reproducible example helps to provide faster answers. If you load data e.g. from a PCD or PLY file, please provide the file.

**Screenshots/Code snippets**

In order to help explain your problem, please consider adding
* screenshots of the GUI issues
* code snippets: [syntax for code](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#code) with correct language highlights

**Your Environment (please complete the following information):**

 - OS: [e.g. Ubuntu 16.04]
 - Compiler: [:eg GCC 8.1]
 - PCL Version 1.12.1

**Possible Solution**

Not obligatory, but suggest a fix/reason for the bug. Feel free to create a PR if you feel comfortable.

**Additional context**

Add any other context about the problem here.

--- a/registration/include/pcl/registration/impl/ia_fpcs.hpp
+++ b/registration/include/pcl/registration/impl/ia_fpcs.hpp
@@ -55,7 +55,7 @@ pcl::getMeanPointDensity(const typename pcl::PointCloud<PointT>::ConstPtr& cloud
                          int nr_threads)
 {
   const float max_dist_sqr = max_dist * max_dist;
-  const std::size_t s = cloud.size();
+  const std::size_t s = cloud->size();
 
   pcl::search::KdTree<PointT> tree;
   tree.setInputCloud(cloud);

